### PR TITLE
[FMV] Document feature dependencies and detect at selection.

### PR DESCRIPTION
**Pull request by labrinea**
---
name: Pull request
about: Technical issues, document format problems, bugs in scripts or feature proposal.

---

**Thank you for submitting a pull request!**

If this PR is about a bugfix:

Please use the bugfix label and make sure to go through the checklist below.

If this PR is about a proposal:

We are looking forward to evaluate your proposal, and if possible to
make it part of the Arm C Language Extension (ACLE) specifications.

We would like to encourage you reading through the [contribution
guidelines](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md), in particular the section on [submitting
a proposal](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md#proposals-for-new-content).

Please use the proposal label.

As for any pull request, please make sure to go through the below
checklist.

Checklist: (mark with ``X`` those which apply)

* [ ] If an issue reporting the bug exists, I have mentioned it in the
      PR (do not bother creating the issue if all you want to do is
      fixing the bug yourself).
* [ ] I have added/updated the `SPDX-FileCopyrightText` lines on top
      of any file I have edited. Format is `SPDX-FileCopyrightText:
      Copyright {year} {entity or name} <{contact informations}>`
      (Please update existing copyright lines if applicable. You can
      specify year ranges with hyphen , as in `2017-2019`, and use
      commas to separate gaps, as in `2018-2020, 2022`).
* [ ] I have updated the `Copyright` section of the sources of the
      specification I have edited (this will show up in the text
      rendered in the PDF and other output format supported). The
      format is the same described in the previous item.
* [ ] I have run the CI scripts (if applicable, as they might be
      tricky to set up on non-*nix machines). The sequence can be
      found in the [contribution
      guidelines](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md#continuous-integration). Don't
      worry if you cannot run these scripts on your machine, your
      patch will be automatically checked in the Actions of the pull
      request.
* [ ] I have added an item that describes the changes I have
      introduced in this PR in the section **Changes for next
      release** of the section **Change Control**/**Document history**
      of the document. Create **Changes for next release** if it does
      not exist. Notice that changes that are not modifying the
      content and rendering of the specifications (both HTML and PDF)
      do not need to be listed.
* [ ] When modifying content and/or its rendering, I have checked the
      correctness of the result in the PDF output (please refer to the
      instructions on [how to build the PDFs
      locally](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md#continuous-integration)).
* [ ] The variable `draftversion` is set to `true` in the YAML header
      of the sources of the specifications I have modified.
* [ ] Please *DO NOT* add my GitHub profile to the list of contributors
      in the [README](https://github.com/ARM-software/acle/blob/main/README.md#contributors-) page of the project.

### CODE REVIEW

**Comment by labrinea** [comment]
@jroelofs @andrewcarlotti @Wilco1

File: main/acle.md [hunk 1]
```diff
@@ -2830,6 +2826,47 @@ The following table lists the architectures feature mapping for AArch64
    | 580           | `FEAT_SME2`              | sme2          | ```ID_AA64PFR1_EL1.SMEver >= 0b0001```    |
    | 650           | `FEAT_MOPS`              | mops          | ```ID_AA64ISAR2_EL1.MOPS >= 0b0001```     |
 
+### Dependencies
+
+If a feature depends on another feature as defined by the table below then
+no need to explicitly state in the attribute. These dependencies are taken
```
Comment (jroelofs) [inline main/acle.md:2832]: I think this should be a little more explicit that both:
*  the depended-on feature _need not_ be specified
*  the depended-on feature _may_ be specified

**Review by jroelofs — APPROVED** [review]

File: main/acle.md [hunk 2]
```diff
@@ -2830,6 +2826,50 @@ The following table lists the architectures feature mapping for AArch64
...
+   | flagm2           | flagm             |
+   | simd             | fp                |
+   | dotprod          | simd              |
+   | sm4              | simd              |
+   | rdm              | simd              |
+   | sha2             | simd              |
+   | sha3             | sha2              |
+   | aes              | simd              |
+   | fp16             | fp                |
+   | fp16fml          | simd, fp16        |
```
Comment (andrewcarlotti) [inline main/acle.md:2851]: Last time I checked, I don't think any part of the toolchains has an fp16fml->simd dependency for command line arguments, so this seems inconsistent.

File: main/acle.md [hunk 3]
```diff
@@ -2830,6 +2826,50 @@ The following table lists the architectures feature mapping for AArch64
...
+   | dpb2             | dpb               |
+   | jscvt            | fp                |
+   | fcma             | simd              |
+   | rcpc2            | rcpc              |
+   | rcpc3            | rcpc2             |
+   | frintts          | fp                |
+   | i8mm             | simd              |
+   | bf16             | simd              |
```
Comment (andrewcarlotti) [inline main/acle.md:2859]: In GCC+Bintuils we only have bf16 depending on fp, and in LLVM it depends on nothing at all. I think fp is more appropriate than simd.

File: main/acle.md [hunk 4]
```diff
@@ -2830,6 +2826,50 @@ The following table lists the architectures feature mapping for AArch64
...
+   | sve              | fp16              |
```
Comment (andrewcarlotti) [inline main/acle.md:2860]: This should probably have a simd dependency, and perhaps also an fcma dependency?

File: main/acle.md [hunk 5]
```diff
@@ -2830,6 +2826,50 @@ The following table lists the architectures feature mapping for AArch64
...
+   | f32mm            | sve               |
+   | f64mm            | sve               |
+   | sve2             | sve               |
+   | sve2-aes         | sve2, aes         |
+   | sve2-bitperm     | sve2              |
+   | sve2-sha3        | sve2, sha3        |
+   | sve2-sm4         | sve2, sm4         |
+   | sme              | fp16, bf16        |
```
Comment (andrewcarlotti) [inline main/acle.md:2868]: This doesn't currently match existing command line options, but those should probably be adjusted. Should this also have a simd and perhaps fcma dependency?

File: main/acle.md [hunk 6]
```diff
@@ -2830,6 +2826,50 @@ The following table lists the architectures feature mapping for AArch64
...
 
+### Dependencies
+
+If a feature depends on another feature as defined by the table below then:
+
+* the depended-on feature *need not* be specified in the attribute,
+* the depended-on feature *may* be specified in the attribute.
+
+These dependencies are taken into account transitively when selecting the
+most appropriate version of a function (see section [Selection](#selection)).
```
Comment (andrewcarlotti) [inline main/acle.md:2837]: I don't agree with this - while it's unlikely to come up in practice, I think the selection algorithm would be simpler to understand if it were based solely upon the attribute strings and the feature priorities, without trying to detect and ignore any redundant dependencies.

File: main/acle.md [hunk 7]
```diff
@@ -2830,6 +2826,50 @@ The following table lists the architectures feature mapping for AArch64
    | 580           | `FEAT_SME2`              | sme2          | ```ID_AA64PFR1_EL1.SMEver >= 0b0001```    |
    | 650           | `FEAT_MOPS`              | mops          | ```ID_AA64ISAR2_EL1.MOPS >= 0b0001```     |
 
+### Dependencies
+
+If a feature depends on another feature as defined by the table below then:
+
+* the depended-on feature *need not* be specified in the attribute,
+* the depended-on feature *may* be specified in the attribute.
```
Comment (philgee-oss) [inline main/acle.md:2834]: *might*

File: main/acle.md [hunk 8]
```diff
@@ -2830,6 +2826,50 @@ The following table lists the architectures feature mapping for AArch64
    | 580           | `FEAT_SME2`              | sme2          | ```ID_AA64PFR1_EL1.SMEver >= 0b0001```    |
    | 650           | `FEAT_MOPS`              | mops          | ```ID_AA64ISAR2_EL1.MOPS >= 0b0001```     |
 
+### Dependencies
+
+If a feature depends on another feature as defined by the table below then:
```
Comment (philgee-oss) [inline main/acle.md:2831]: the following table then

**Pull request merged**